src/control/lib/netdetect/netdetect.go

```diff
@@ -177,3 +557,276 @@ func GetDeviceNames() ([]string, error) {
+		return "psm2", nil
+	case "ofi+gni":
+		return "gni", nil
+	default:
+	}
+	return "", errors.Errorf("fabric provider: %s not known by libfabric.  Use 'daos_server network list' to view supported providers", provider)
+}
+
+// ConvertMercuryToLibFabric converts a Mercury provider string containing one or more providers
+// separated by ';' into a libfabric compatible provider string
+// All subproviders must convert successfully in order to have a successful conversion.
+func ConvertMercuryToLibFabric(provider string) (string, error) {
+	var libFabricProviderList string
+
+	if len(provider) == 0 {
+		return "", errors.New("fabric provider was empty")
+	}
+
+	tmp := strings.Split(provider, ";")
+	for _, subProvider := range tmp {
+		libFabricProvider, err := MercuryToLibFabric(subProvider)
+		if err != nil {
+			return "", errors.Errorf("fabric provider: '%s' is not known by libfabric.  Use 'daos_server network list' to view supported providers", subProvider)
+		}
+		libFabricProviderList += libFabricProvider + ";"
+	}
+	return strings.TrimSuffix(libFabricProviderList, ";"), nil
+}
+
+// LibFabricToMercury converts a single libfabric provider string into a Mercury compatible provider string
+func LibFabricToMercury(provider string) (string, error) {
+	switch provider {
+	case "sockets":
+		return "ofi+sockets", nil
+	case "tcp":
+		return "ofi+tcp", nil
+	case "verbs":
+		return "ofi+verbs", nil
+	case "ofi_rxm":
+		return "ofi_rxm", nil
+	case "psm2":
+		return "ofi+psm2", nil
+	case "gni":
+		return "ofi+gni", nil
+	default:
+	}
+
+	return "", errors.Errorf("fabric provider: %s not known by Mercury", provider)
+}
+
+// ConvertLibFabricToMercury converts a libfabric provider string containing one or more providers
+// separated by ';' into a Mercury compatible provider string
+// At least one provider in the list must convert to a Mercury provider in order to have success.
+func ConvertLibFabricToMercury(provider string) (string, error) {
+	var mercuryProviderList string
+
+	log.Debugf("Converting provider string: '%s' to Mercury", provider)
+
+	if len(provider) == 0 {
+		return "", errors.New("fabric provider was empty.")
+	}
+
+	tmp := strings.Split(provider, ";")
+	for _, subProvider := range tmp {
+		mercuryProvider, err := LibFabricToMercury(subProvider)
+		// It's non-fatal if LibFabricToMercury returns an error.  It just means that
+		// this individual provider could not be converted.  It is only an error when
+		// none of the providers can be converted.
+		if err != nil {
+			continue
+		}
+		mercuryProviderList += mercuryProvider + ";"
+	}
+
+	// Success if we converted at least one provider
+	if mercuryProviderList != "" {
+		return strings.TrimSuffix(mercuryProviderList, ";"), nil
+	}
+
+	return "", errors.Errorf("failed to convert the libfabric provider list '%s' to any mercury provider", provider)
+}
+
+// ValidateNetworkConfigStub is used for most unit testing to replace ValidateNetworkConfig because the network configuration
```

Comment:
this should be in a _test.go file IMO

Reply:
I consulted with Mike to get his opinion.  He agreed that it was OK to keep this as-is.